source/modulo_interfaces/msg/JointPositionsCollection.msg

```diff
@@ -0,0 +1,3 @@
+std_msgs/Header header
```

Comment:
I'm actually not sure about this, should the header not rather be in the JointPositions message or not be there at all? Usually, clients will not want to create that message so if we want to work with a timestamp then it should be associated with a joint positions and not with a collection

Reply:
to me personally this seemed natural, but indeed the `JointPositions.msg` then has nothing if we end up using it more. But then when using `JointPositionsCollection` each joint position would carry its own header which would be identical for all?

I intuitively expect the top-most struct to carry the header.

I sometimes feel like we miss a bit on the ROS-like way which would be duplicate the message and have a `JointPositionsStamped`. Would it make sense to add the latter and then have the JointPositionsCollection have `JointPositions` messages and use `JointPositionsStamped` for other things?

Reply:
that's an option.

If we look at TF (which is the role model for this feature), then the message type of `/tf` is TFMessage which is simply a vector of transform messages that each have a header (because every TF should have it's stamp). The `TFMessage` is just a collection of those such that you can do batch publishing and not every TF by itself. That's why maybe that logic makes more sense for us too. However, we have to keep in mind that what we do here is all "static" and hence the time maybe doesn't count as much anyway.

Reply:
I see the reasoning here and overall it would be fine if each JointPositions message has a header. Even though they are "static" we might still be interested which ones are published when.

But, our predicate messages also don't have a timestamp so it's also fine to me to just leave the header out entirely.

I don't think the intermediate option of having multiple messages, stamped or unstamped, is particularly useful just because we will probably end up using only one or the other.